src/Microsoft.Extensions.PlatformAbstractions/DefaultApplicationEnvironment.cs

```diff
@@ -55,6 +34,18 @@ public FrameworkName RuntimeFramework
             }
         }
 
+        private static string GetApplicationBasePath()
+        {
+            var basePath =
+#if NET451
+                (string)AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY") ??
+                AppDomain.CurrentDomain.BaseDirectory;
+#else
+                AppContext.BaseDirectory;
+#endif
+            return Path.GetFullPath(basePath);
```

Comment:
Does this do what we think it does? Can you print out what this is for a vanilla mono app compared to AppDomain.CurrentDomain.BaseDirectory ?